pkg/volume/csi/csi_block.go

```diff
@@ -331,31 +373,96 @@ func (m *csiBlockMapper) TearDownDevice(globalMapPath, devicePath string) error
 		return errors.New(log("blockMapper.TearDownDevice failed to get CSI client: %v", err))
 	}
 
-	// Call NodeUnpublishVolume
-	publishPath := m.getPublishPath()
-	if _, err := os.Stat(publishPath); err != nil {
+	// Call NodeUnstageVolume
+	stagingPath := m.getStagingPath()
+	if _, err := os.Stat(stagingPath); err != nil {
 		if os.IsNotExist(err) {
-			klog.V(4).Infof(log("blockMapper.TearDownDevice publishPath(%s) has already been deleted, skip calling NodeUnpublishVolume", publishPath))
+			klog.V(4).Infof(log("blockMapper.TearDownDevice stagingPath(%s) has already been deleted, skip calling NodeUnstageVolume", stagingPath))
 		} else {
 			return err
 		}
 	} else {
-		err := m.unpublishVolumeForBlock(ctx, csiClient, publishPath)
+		err := m.unstageVolumeForBlock(ctx, csiClient, stagingPath)
 		if err != nil {
 			return err
 		}
 	}
 
-	// Call NodeUnstageVolume
-	stagingPath := m.getStagingPath()
-	if _, err := os.Stat(stagingPath); err != nil {
+	dataDir := getVolumeDeviceDataDir(m.specName, m.plugin.host)
```

Comment:
this change was just reverted :/ https://github.com/kubernetes/kubernetes/pull/83379

Reply:
Thank you for quickly letting me know it!